api/API_VERSIONING.md

```diff
@@ -28,9 +29,24 @@ configuration resources (e.g. bootstrap, xDS resources such as `Cluster`). The
 v3 API bootstrap configuration is `envoy.config.bootstrap.v3.Boostrap`, even
 though it might might transitively reference `envoy.service.trace.v2`.
 
+The minor version enables API stability and support, and to safely deprecate old
+features and reduce the tech debt. A client can support a range of minor API
+versions, and can request xDS resources that match this range, using the resource
+[context parameters](https://github.com/cncf/udpa/blob/master/xds/core/v3/context_params.proto).
+Deprecated features, from previous minor or major versions (and their code), can be safely removed
+from the client. In general, within a minor API version, we do not allow any breaking changes
+(more details below). The minor version will be reset to 0 upon a release of a new major version.
+
+The patch version tracks incremental changes to the API, such as, adding a new
+package, adding a new field, deprecating a field, etc.
+A client can supply the supported patch version when requesting xDS resources,
+which allows clients and servers to support new features, before the release of
```

Comment:
I'd be more crisp. It's already possible for servers to unconditionally set new fields and have some clients support and others ignore. What's not possible today is for a server to detect if a client supports a newer field without patch version. This is a way for servers to selectively condition there response based on whether a client has a new feature or not.

Reply:
Rephrased.